File: leetcode/cpp/1009_Complement_of_Base_10_Integer.cpp
class Solution {
public:
	int bitwiseComplement(int n) {
		if (n == 0) return 1;
        int msb = 31 - __builtin_clz(n);
		// pushes 10000msb.... and the -1
		// gives us 01111msb...
        int mask = ~((1 << (msb + 1)) - 1);
        int result = n | mask;
        return ~result;
	}
};

File: leetcode/cpp/1009_Complement_of_Base_10_Integer.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        if n == 0:
            return 1

        msb = n.bit_length() - 1

        mask = ~((1 << (msb + 1)) - 1)

        result = n | mask

        return ~result
